### PR TITLE
Fix several things around root systems

### PR DESCRIPTION
**Pull request by lgoettgens**
While writing a jupyter notebook for an exercise class, I found several things that weren't working or just weren't there. This is the first batch of fixes and additions from that.

### CODE REVIEW

**Review by simonbrandhorst — APPROVED** [review]

**Comment by lgoettgens** [comment]
@felix-roehrich please review the changes here, even though it has already been merged. I can react to your comments in a future PR.

File: experimental/LieAlgebras/src/RootSystem.jl [hunk 1]
```diff
@@ -164,6 +175,15 @@ function roots(R::RootSystem)
   return [[r for r in positive_roots(R)]; [-r for r in positive_roots(R)]]
 end
 
+function simple_root(R::RootSystem, i::Int)
+  @req 1 <= i <= rank(R) "Invalid index"
+  return positive_root(R, i)
+end
+
+function simple_roots(R::RootSystem)
+  return [positive_root(R, i) for i in 1:num_simple_roots(R)]
```
Comment (felix-roehrich) [inline experimental/LieAlgebras/src/RootSystem.jl:184]:
```suggestion
  return positive_roots(R)[1:rank(R)]
```

File: experimental/LieAlgebras/src/WeylGroup.jl [hunk 2]
```diff
@@ -141,6 +141,24 @@ function Base.:(*)(x::WeylGroupElem, w::WeightLatticeElem)
...
+  end
+
+  word = deepcopy(x.word)
+  for _ in 2:n
+    for s in Iterators.reverse(x.word)
+      _lmul!(x.parent.refl, word, s)
+    end
+  end
+
+  return WeylGroupElem(x.parent, word)
```
Comment (felix-roehrich) [inline experimental/LieAlgebras/src/WeylGroup.jl:159]:
```suggestion
  px = deepcopy(x)
  for _ in 2:n
    for s in Iterators.reverse(x.word)
      lmul!(px, Int(s)) # make lmul! more general to make casting unnecessary
    end
  end

  return px
```

File: experimental/LieAlgebras/src/RootSystem.jl (hunk 1 above)
Comment (lgoettgens) [reply experimental/LieAlgebras/src/RootSystem.jl:184]: The current solution is more efficient than your suggestion. Your suggestion first creates a vector of all positive roots and then copies the first `rk` many to a new vector. As there are many more positive than simple roots, this is mostly useless work.

Comment (felix-roehrich) [reply experimental/LieAlgebras/src/RootSystem.jl:184]: No, the implementation is just an accessor with a type assertion. The suggestion instead prevents creating a new underlying array each time.

Comment (lgoettgens) [reply experimental/LieAlgebras/src/RootSystem.jl:184]: Oh yeah, you are correct.

**Pull request merged**